### PR TITLE
event subscription implemented

### LINKED ISSUE #357: implement event subscriptions
create following files

 - `/src/events/users`
 - `/src/events/demographics` 

they should export functions like

`eventUserCreated(userId)`
`eventUserUpdated(userId)`
`eventDemographicsCreated(demoId, userId)`

these should be async functions so that they can run non blocking. 

the purpose of these functions would be like this - 

```js
function eventUserCreated(userId) {
   // find all EventSubscriptions that exist for {user, created}
   // make a POST request to all the webhookURL for all such clients
}
```

payload of post request should have nominal data like ids 

```
{
  type: "CREATED",
  model: "user",
  success: true,
  id: 1,
  userId: 1   
}
```
<small> in case of user model events, id and userId will be same only</small>

```
{
  type: "UPDATED",
  model: "address",
  success: true,
  id: 10,
  userId: 1

}
```

these functions like `eventUserCreated` should be called from controller for user creation. 

The controller should **NOT** await for event function. The event is fired and forgotten, if the event is not reported, the controller should not get stuck for that.

### PR DESCRIPTION
**Pull request by Manit1**
Fixes #357

### CODE REVIEW

**Comment by championswimmer** [comment]
Please remove yarn.lock
We prefer to run via npm only for now, and keep only package-lock.json

File: src/utils/subscriptions.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      model,
+      type
+    }
+  })
+}
+
+function sendEvent(subscriptions, model, type, id, userId) {
+  return Promise.all(subscriptions.map(async subscription => {
+    const client = await Client.findById(subscription.clientId)
+    const webhookURL = client.webhookURL
```
Comment (championswimmer) [inline src/utils/subscriptions.js]: what if client's webhookURL is null ?

Comment (Manit1) [reply src/utils/subscriptions.js]: Done !

Comment (Manit1) [reply src/utils/subscriptions.js]: @championswimmer Done !

**Comment by championswimmer** [comment]
This pull request **introduces 1 alert** when merging b09822005b98b53e85f90b5e11eb31d9673842ff into 2710b63457ead26bf53685ff77fad086e9abbf5c - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/oneauth/rev/pr-a2fdf8e849f7715e6660f8afe488ada89fd4c242)

**new alerts:**

* 1 for Duplicate character in character class

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by championswimmer** [comment]
This looks good to me ! 
But... 

The UI for creating EventSubscriptions  for the clients needs to be there too right ? 
If I made a client, I should be able to enable subscriptions for my client too (which will fire a `EventSubscriptions.create()` internally) 

This is part of the work (can merge this now, allocate 2000 points, or wait for that to be done and allocate the full 4000)

**Comment by Manit1** [comment]
@championswimmer I am working on the UI part as well but it is listed as a separate issue! That's why I didn't included that here

**Pull request merged**